sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScanExec.scala

```diff
@@ -38,7 +38,10 @@ private[sql] case class LocalTableScanExec(
     rows.map(r => proj(r).copy()).toArray
   }
 
-  private lazy val rdd = sqlContext.sparkContext.parallelize(unsafeRows)
+  private lazy val numParallelism: Int = math.min(math.max(unsafeRows.size, 1),
```

Comment:
Nit: prefer `unsafeRows.length` to `unsafeRows.size`, since the latter involves an implicit conversion.